File: src/Codeforces/educational_round_37/C.py
"""
6
1 2 5 3 4 6
01110
"""


def sort_this_part(start, end):
    so = sorted(arr)
    arr[start:end + 1] = so[start:end + 1]


if __name__ == '__main__':
    n = int(input().strip())
    arr = [int(i) for i in input().strip().split()]
    l = list(input().strip()) + ['0']
    i = 0
    while i < n - 1:
        if l[i] == '1':
            start = i
            while l[i] == '1' and i < n - 1:
                i += 1
            i += 1
            sort_this_part(start, i)
        else:
            i += 1
    print("YES" if arr == sorted(arr) else "NO")

File: src/Codeforces/educational_round_37/C.java
package Codeforces.educational_round_37;

/*
 * Created by bk on 04-02-2018 00:05
 */

import java.util.Arrays;
import java.util.Scanner;

public class C {
    public static void main(String[] __) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) arr[i] = sc.nextInt();
        String valid = sc.next();
        int[] l = new int[n];
        for (int i = 0; i < n - 1; i++) l[i] = valid.charAt(i) == '1' ? 1 : 0;
        int i = 0;
        while (i < n - 1) {
//            System.out.println(Arrays.toString(arr));
            if (l[i] == 1) {
                int start = i;
                while (l[i] == 1 && i < n - 1) i += 1;
                Arrays.sort(arr, start, i + 1);
            } else i += 1;
        }
        boolean correct = true;
        for (i = 0; i < n; i++) {
            if (arr[i] != i + 1) correct = false;
            if (!correct) break;
        }
        System.out.println(correct ? "YES" : "NO");
    }
}
